browser/extensions/api/moonpay_api.cc

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/extensions/api/moonpay_api.h"
+
+#include <string>
+#include <memory>
+
+#include "base/environment.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/moonpay/browser/regions.h"
+#include "brave/components/moonpay/common/moonpay_pref_names.h"
+#include "brave/components/ntp_widget_utils/browser/ntp_widget_utils_region.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/prefs/pref_service.h"
+
+namespace extensions {
+namespace api {
+
+ExtensionFunction::ResponseAction
+MoonpayIsBitcoinDotComSupportedFunction::Run() {
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+
+  if (brave::IsTorProfile(profile)) {
```

Comment:
I think it would be better to have something like moonpay::IsSupportedProfile because I think we also want to exclude guests and maybe some others. cc @yrliou because I think we might have a more generic method that covers all of those

Reply:
`browser/profiles/profile_util.cc:bool`'s `IsRegularProfile`

Reply:
done